astronomy-pipeline/extract_functions.py

```diff
@@ -0,0 +1,185 @@
+"""File to obtain body positional data from the Astronomy API."""
+
+# pylint: disable=E0401,R0913,R0917
+
+from os import environ as ENV
+from datetime import date, timedelta
+import base64
+
+
+from psycopg2 import connect, extensions, extras
+from dotenv import load_dotenv
+import requests
+
+load_dotenv()
+
+ASTRO_URL = "https://api.astronomyapi.com/api/v2"
+
+
+def get_db_connection() -> extensions.connection:
+    """Reusable function for getting a database connection."""
+
+    return connect(cursor_factory=extras.RealDictCursor,
+                   dbname=ENV["DB_NAME"],
+                   host=ENV["DB_HOST"],
+                   password=ENV["DB_PASSWORD"],
+                   user=ENV["DB_USER"],
+                   port=ENV["DB_PORT"])
+
+
+def get_auth_string() -> str:
```

Comment:
Interacting wit the astronomyAPI requires an authorisation key created from generated secrets. Nice function that accounts for this, LGTM.